src/webgl/material.js

```diff
@@ -718,171 +725,521 @@ p5.prototype.createFilterShader = function (fragSrc) {
+ *   fillShader.setUniform('uLightPos', lightPos); // Mouse changes light position
  *   plane(100, 100);
- *
- *   describe('A yellow square.');
  * }
  * </code>
  * </div>
  *
- * <div>
+ * @example
+ * <div modernizr='webgl'>
  * <code>
- * // Note: A "uniform" is a global variable within a shader program.
+ * // Example 3: Mixed Fill Shader with Lights and No Lights
+ * // Light interaction with fill shader
+ * let fillShader;
  *
- * let mandelbrot;
+ * let vertSrc = `
+ * precision highp float;
+ * attribute vec3 aPosition;
+ * uniform mat4 uProjectionMatrix;
+ * uniform mat4 uModelViewMatrix;
+ * varying vec3 vPosition;
+ * void main() {
+ *   vPosition = aPosition;
+ *   gl_Position = uProjectionMatrix * uModelViewMatrix * vec4(aPosition, 1.0);
+ * }
+ * `;
  *
- * // Load the shader and create a p5.Shader object.
- * function preload() {
- *   mandelbrot = loadShader('assets/shader.vert', 'assets/shader.frag');
+ * let fragSrc = `
+ * precision highp float;
+ * uniform vec3 uLightPos;
+ * uniform vec3 uFillColor;
+ * varying vec3 vPosition;
+ * void main() {
+ *   float brightness = dot(normalize(uLightPos), normalize(vPosition));
+ *   brightness = clamp(brightness, 0.0, 1.0);
+ *   vec3 color = uFillColor * brightness;
+ *   gl_FragColor = vec4(color, 1.0);
  * }
+ * `;
  *
  * function setup() {
  *   createCanvas(100, 100, WEBGL);
+ *   fillShader = createShader(vertSrc, fragSrc);
+ *   shader(fillShader);
+ *   noStroke();
+ *   describe('A square affected by both fill color and lighting, with lights controlled by mouse.');
+ * }
  *
- *   // Use the p5.Shader object.
- *   shader(mandelbrot);
+ * function draw() {
+ *   let lightPos = [(mouseX - width / 2) / width, (mouseY - height / 2) / height, 1.0];
+ *   fillShader.setUniform('uLightPos', lightPos);
+ *   let fillColor = [map(mouseX, 0, width, 0, 1), map(mouseY, 0, height, 0, 1), 0.5];
+ *   fillShader.setUniform('uFillColor', fillColor);
+ *   plane(100, 100);
+ * }
+ * </code>
+ * </div>
+ */
+p5.prototype.shader = function (s) {
+  this._assert3d('shader');
+  p5._validateParameters('shader', arguments);
+
+  s.ensureCompiledOnContext(this);
+
+  // Always set the shader as a fill shader
+  this._renderer.userFillShader = s;
+  this._renderer._useNormalMaterial = false;
+
+  return this;
+};
+
+/**
+ * Sets the <a href="#/p5.Shader">p5.Shader</a> object to apply for strokes.
  *
- *   // Set the shader uniform p to an array.
- *   mandelbrot.setUniform('p', [-0.74364388703, 0.13182590421]);
+ * This method applies the given shader to strokes, allowing customization of
+ * how lines and outlines are drawn in 3D space. The shader will be used for
+ * strokes until <a href="#/p5/resetShader">resetShader()</a> is called or another
+ * strokeShader is applied.
  *
- *   describe('A fractal image zooms in and out of focus.');
+ * The shader will be used for:
+ * - Strokes only, regardless of whether the uniform `uStrokeWeight` is present.
```

Comment:
mmm....that sounds a bit odd. Specially for the new user learning about `strokeShader`, they will think why this is written if they have no idea working with older `shader()` function for strokes. Can we ommit this line with `uStrokeWeight` uniform?

Also, I see you have three examples of `strokeShader`, as I mentioned earlier, we could have one with shaderHooks (you can see the refrence here : https://github.com/processing/p5.js/pull/7149#issuecomment-2336404893 made by @davepagurek )  and one could be the last example you have drawn currently.  Otherwise looks good.